ir/src/test/scala/org/scalajs/core/ir/PrintersTest.scala

```diff
@@ -0,0 +1,999 @@
+  }
+
+  @Test def printIf(): Unit = {
+    assertPrintEquals(
+        """
+          |if (true) {
+          |  5
+          |} else {
+          |  6
+          |}
+        """,
+        If(b(true), i(5), i(6))(IntType))
+
+    assertPrintEquals(
+        """
+          |if (true) {
+          |  5
+          |}
+        """,
+        If(b(true), i(5), Skip())(NoType))
+
+    assertPrintEquals(
+        """
+          |if (true) {
+          |  5
+          |} else if (false) {
+          |  6
+          |} else {
+          |  7
+          |}
+        """,
+        If(b(true), i(5), If(b(false), i(6), i(7))(IntType))(IntType))
+
+    assertPrintEquals("x || y",
+        If(ref("x", BooleanType), b(true), ref("y", BooleanType))(BooleanType))
+
+    assertPrintEquals("x && y",
+        If(ref("x", BooleanType), ref("y", BooleanType), b(false))(BooleanType))
+  }
+
+  @Test def printWhile(): Unit = {
+    assertPrintEquals(
+        """
+          |while (true) {
+          |  5
+          |}
+        """,
+        While(b(true), i(5)))
+
+    assertPrintEquals(
+        """
+          |lab: while (true) {
+          |  5
+          |}
+        """,
+        While(b(true), i(5), Some("lab")))
+  }
+
+  @Test def printDoWhile(): Unit = {
+    assertPrintEquals(
+        """
+          |do {
+          |  5
+          |} while (true)
+        """,
+        DoWhile(i(5), b(true)))
+
+    assertPrintEquals(
+        """
+          |lab: do {
+          |  5
+          |} while (true)
+        """,
+        DoWhile(i(5), b(true), Some("lab")))
+  }
+
+  @Test def printTry(): Unit = {
+    assertPrintEquals(
+        """
+          |try {
+          |  5
+          |} catch (e) {
+          |  6
+          |}
+        """,
+        Try(i(5), "e", i(6), EmptyTree)(IntType))
+
+    assertPrintEquals(
+        """
+          |try {
+          |  5
+          |} finally {
+          |  6
+          |}
+        """,
+        Try(i(5), "e", EmptyTree, i(6))(IntType))
+
+    assertPrintEquals(
+        """
+          |try {
+          |  5
+          |} catch (e) {
+          |  6
+          |} finally {
+          |  7
+          |}
+        """,
+        Try(i(5), "e", i(6), i(7))(IntType))
+  }
+
+  @Test def printThrow(): Unit = {
+    assertPrintEquals("throw null", Throw(Null()))
+  }
+
+  @Test def printContinue(): Unit = {
+    assertPrintEquals("continue", Continue())
+    assertPrintEquals("continue lab", Continue(Some("lab")))
+  }
+
+  @Test def printMatch(): Unit = {
+    assertPrintEquals(
+        """
+          |match (x) {
+          |  case 5:
+          |    6;
+          |  case 7 | 8:
+          |    {
+          |      9;
+          |      10
+          |    };
+          |  default:
+          |    11;
+          |}
+        """,
+        Match(ref("x", IntType), List(
+            List(i(5)) -> i(6),
+            List(i(7), i(8)) -> Block(i(9), i(10))),
+            i(11))(IntType))
```

Comment:
No. There are other cases where we do not specialize `skip`s, e.g., the body of `while` loops. This should not happen often, so we don't bother.

The cases we do specialize are things that happen often in typical Scala code. An empty default is very rare.